tests/models/hydrology/test_hydrology_tools.py

```diff
@@ -31,80 +16,107 @@ def test_setup_hydrology_input_current_timestep(
     var_list = [
         "current_precipitation",
-        "subcanopy_temperature",
-        "subcanopy_humidity",
-        "subcanopy_pressure",
-        "subcanopy_wind_speed",
-        "leaf_area_index_sum"
-        "current_evapotranspiration"
-        "soil_layer_heights"
-        "soil_layer_thickness"
-        "top_soil_moisture_capacity_mm"
-        "top_soil_moisture_residual_mm"
-        "soil_moisture_true",
-        "previous_accumulated_runoff"
-        "previous_subsurface_flow_accumulated"
+        # "subcanopy_temperature",
+        # "subcanopy_humidity",
+        # "subcanopy_pressure",
+        # "subcanopy_wind_speed",
+        "surface_temperature",
+        "surface_humidity",
+        "surface_pressure",
+        "surface_wind_speed",
+        "leaf_area_index_sum",
+        "current_evapotranspiration",
+        # "soil_layer_heights",
+        # "soil_layer_thickness",
+        "top_soil_moisture_capacity_mm",
+        "top_soil_moisture_residual_mm",
+        # "soil_moisture_true",
+        "previous_accumulated_runoff",
+        "previous_subsurface_flow_accumulated",
         "groundwater_storage",
+        "soil_moisture_mm",
+        "latent_heat_vapourisation",
+        "molar_density_air",
     ]
 
-    variables = [var for var in result if var not in var_list]
-    assert variables
+    assert set(result.keys()) == set(var_list)
 
     # check if climate values are selected correctly
     np.testing.assert_allclose(
         np.sum(result["current_precipitation"], axis=1),
         (dummy_climate_data["precipitation"].isel(time_index=0)).to_numpy(),
     )
     np.testing.assert_allclose(
-        result["subcanopy_temperature"], dummy_climate_data["air_temperature"][11]
+        result["surface_temperature"],
+        dummy_climate_data["air_temperature"]
+        .isel(layers=fixture_core_components.layer_structure.role_indices["surface"])
+        .squeeze(),
     )
     np.testing.assert_allclose(
-        result["subcanopy_humidity"], dummy_climate_data["relative_humidity"][11]
+        result["surface_humidity"],
+        dummy_climate_data["relative_humidity"]
+        .isel(layers=fixture_core_components.layer_structure.role_indices["surface"])
+        .squeeze(),
     )
+    # The reference data is a time series with cell id in axis 0, the result has cell_id
+    # on axis 1, so need to extract as 2d and rotate to match.
     np.testing.assert_allclose(
-        result["subcanopy_pressure"],
-        (dummy_climate_data["atmospheric_pressure_ref"].isel(time_index=0)).to_numpy(),
+        result["surface_pressure"],
+        dummy_climate_data["atmospheric_pressure_ref"]
+        .isel(time_index=[0])
+        .squeeze()
+        .to_numpy()
+        .T,
     )
     np.testing.assert_allclose(
-        result["soil_moisture_mm"], DataArray(np.full((2, 3), 50.0))
+        result["soil_moisture_mm"],
+        DataArray(np.tile([[5], [500]], fixture_core_components.grid.n_cells)),
     )
```

Comment:
Again, this should not be returned, could this have to do with writing directly in the output dict without renaming the variable?